internal/bminventory/inventory.go

```diff
@@ -2078,22 +2074,17 @@ func optionalParam(data *string, field string, updates map[string]interface{}) {
 }
 
 func (b *bareMetalInventory) updateHostRoles(ctx context.Context, params installer.UpdateClusterParams, db *gorm.DB, log logrus.FieldLogger) error {
-
 	for i := range params.ClusterUpdateParams.HostsRoles {
-		log.Infof("Update host %s to role: %s", params.ClusterUpdateParams.HostsRoles[i].ID,
-			params.ClusterUpdateParams.HostsRoles[i].Role)
-		var host models.Host
-		err := db.First(&host, "id = ? and cluster_id = ?",
-			params.ClusterUpdateParams.HostsRoles[i].ID, params.ClusterID).Error
+		hostRole := params.ClusterUpdateParams.HostsRoles[i]
+		log.Infof("Update host %s to role: %s", hostRole.ID, hostRole.Role)
+		var host, err = getHost(hostRole.ID.String(), params.ClusterID.String(), db, log)
```

Comment:
Same for this change - it's a nice refactor but doesn't seem relevant to the PR / commit message